congestion-control-charter.txt

```diff
@@ -1,85 +1,34 @@
-if they meet the criteria.
-
-* Real-time, media adaptation algorithms for peer to peer communications
-
-* Operational guidance, in general, including advice to upper layers or new
-transport protocols. Examples include minimum frequency of feedback,
-counterproductive application retransmissions and multiple connections to the
-same host.
-
-Proposals that depend on the capabilities of a single transport protocol should
-generally remain in the maintenance working group for that protocol (i.e.,
-TCPM, QUIC, TSVWG). Documents in CONGRESS should remain as independent of
-transport protocol as possible, but they might have short specific instructions
-(e.g. a header option or parameter format) for one or more protocols.
-
-CONGRESS is not chartered to publish Informational RFCs documenting the state of
-congestion control in the Internet, including assessments of compliance with
-existing standards. Other venues, e.g., IRTF, may be more appropriate for
-publishing such documents.
-
-CONGRESS will not remain open simply “in case” further work comes along. 
+Congestion Control Working Group Charter (CCWG)
+
+
+RFC 5033 describes a Best Current Practice to evaluate new congestion control
+algorithms as Experimental or Proposed Standard RFCs. TCP was the dominant
+consumer of this work, and proposals were typically discussed in research
+groups (for example, the Internet Congestion Control Research Group - ICCRG).
+
+Since RFC 5033 was published, many conditions have changed. Congestion control
+algorithm proponents now often have the opportunity to test and deploy at scale
+without IETF review. The set of protocols using these algorithms has spread
+beyond TCP and SCTP to include DCCP, QUIC, and beyond. There is more interest
+in specialized use cases such as data centers and real-time protocols. Finally,
+the community has gained much more experience with indications of congestion
+beyond packet loss.
+
+The Congestion Control Working Group will review some of the impediments to 
+congestion control work occurring in the IETF and can generalize transports
+from TCP to all of the relevant transport protocols. This will inform a revision
+of RFC 5033 that encourages IETF review of congestion control proposals and
+standardization of mature congestion control algorithms.
+
+The congestion control expertise in the working group would also make it a
+natural venue to take on other work related to indications of congestion (such
+as delay), queueing algorithms, rate pacing, interaction with other layers,
+etc. In particular, it could address congestion control algorithms with
+empirical evidence of safety and stated intent to deploy by major
+implementations. The working group is intended to be a home for such work, should
+it arise, if not suitable for the IRTF. It is chartered to consider proposals in
+this space, and given consensus to adopt, seek an immediate recharter regardless
+of the status of the initial deliverable.

```

Comment:
These paragraph needs some more work I think. This is not clear to me what we are trying say clearly here. Like the meaning of "it could address" is not very clear. Will we be just discussing the feasibility of publishing focusing on safety and deployment and then recharter to actually publish something? or will also host the demonstration of algorithm efficiency of the particular algorithms which now usually hosted by ICCRG. 

To me one of the purpose of creating the group is to able to publish standard track congestion control algorithms. I think we need to be clear about that intention in the charter. Otherwise, we just do the -bis work and after the -bis work done we close or recharter to publish std track congestion control algorithm.

Reply:
It had sounded like the ask was to do the latter: do the -bis work and then recharter, but I would also be okay if we wanted to include more of that additional context up front, would probably help to not be solely focused on the -bis and instead have some real world test cases for the procedures outlined there.

Reply:
Agree with Eric. Given that we have a few months until IETF 117, perhaps there's an opportunity to see if we can get more concrete work items (on the horizon), and then add related text to the charter.
Most importantly, I would like the charter to be clear that we're interested in **solutions** (e.g., algorithms), and not limit ourselves to procedures, problem statements, or theoretical frameworks.

I'd be happy to land this PR now, and then iterate and add text related to the concrete work on the horizon, but be sure to not bring back the long list of ideas without concrete work on the horizon.

Reply:
What does "if not suitable for IRTF" means. Does this wg only takes solution which is deployed and tested in the wild?

Reply:
Updated to be more clear about the initial litmus test: empirical evidence of safety and stated intent to deploy by major implementations. Deployment and testing in the wild is one way to provide empirical evidence of safety, yes.